src/events/eventToAction.js

```diff
@@ -90,6 +90,7 @@ const actionTypeOfEventType = {
 // This FlowFixMe is because this function encodes a large number of
 // assumptions about the events the server sends, and doesn't check them.
 export default (state: PerAccountState, event: $FlowFixMe): EventAction | null => {
+  const zulipFeatureLevel = getZulipFeatureLevel(state);
```

Comment:
Oh, and I wanted to ask about this added use of the `PerAccountState`. Technically it's discouraged by the jsdoc:

```
/**
 * Translate a Zulip event from the server into one of our Redux actions.
 *
 * If the action is one we don't currently handle, return null.
 * If it's one we don't recognize at all, log a warning and return null.
 *
 * For reference on the events in the Zulip API, see:
 *   https://zulip.com/api/get-events
 *
 * This function takes the Redux state as an argument because for a handful
 * of types of events, we have it attach some pieces of the state inside the
 * resulting action.  That is a now-obsolete workaround for letting our
 * Redux sub-reducers use data from elsewhere in the Redux state; don't add
 * new uses.
 *
 * The new approach is that we pass the global Redux state to each
 * sub-reducer, and they should use that instead.  See ef251f48a for
 * discussion, and a2000b9c8 and its parent for an example of using it.
 */
```

But I think in this case it would be worse to move this condition into the reducer; it would be less at-the-edge for crunchy shell. What do you think?

Reply:
Although, the messages reducer is the logical first place to look for verifying that certain data doesn't get written to the messages state.

Reply:
Yeah, I think this is fine. That text is really written with the add-some-state-fragments-to-action hack in mind; as written it does apply to conditioning on the feature level like this, but I wasn't thinking of that when I wrote it.

The `message` events in particular already get munged here in a couple of other ways: moving the flags, and converting the avatar URL. So that also goes in favor of doing this manipulation here, to concentrate them at one layer.